contracts/tips/src/utils.rs

```diff
@@ -0,0 +1,75 @@
+use cosmwasm_std::Coin;
+use std::collections::btree_map::BTreeMap;
+
+/// Iterates over the coins vector and merges the coins having the same `denom`.
+/// * `coins` - Vector of coins to merge.
+pub fn merge_coins(coins: Vec<Coin>) -> Vec<Coin> {
+    if coins.len() <= 1 {
+        return coins;
+    }
+
+    let mut map: BTreeMap<String, u128> = BTreeMap::new();
```

Comment:
Can we use HashMap, instead?
BTreeMap is for the situation when the keys need to be sorted, see [reference](https://doc.rust-lang.org/std/collections/index.html#use-a-btreemap-when).